keras/backend.py

```diff
@@ -5540,7 +5542,9 @@ def categorical_crossentropy(target, output, from_logits=False, axis=-1):
         )
 
     # scale preds so that the class probas of each sample sum to 1
-    output = output / tf.reduce_sum(output, axis, True)
+    if forced_rescale:
```

Comment:
this conditional check can be removed with the added arg

Reply:
Agree, we can remove the arg and the conditional. One thing to note is that the change might be backwards incompatible for folks who mistakenly had non-softmax inputs to the function without passing `from_logits=True`. We need to be careful with the rollout.

Reply:
@fchollet - we can add the check and raise a warning here - https://github.com/keras-team/keras/blob/master/keras/backend.py#:~:text=def%20_get_logits(output%2C%20from_logits%2C%20op_type%2C%20fn_name)%3A 
wdyt?